bluebrain/repo-patches/packages/neuron/package.py

```diff
@@ -66,180 +48,123 @@ class Neuron(CMakePackage):
-                "+tests",
-            ]
-        ]
-
-        if spec.satisfies("@:8"):
-            args.append("-DNRN_ENABLE_BINARY_SPECIAL=ON")
-
-        # dynamic mpi support not possible in case of gpu support
-        if "+mpi" in spec:
-            dynamic_mpi = "OFF" if "+gpu" in spec else "ON"
-            args.append("-DNRN_ENABLE_MPI_DYNAMIC=%s" % dynamic_mpi)
-
-        if "+python" in spec:
-            args.append("-DPYTHON_EXECUTABLE:FILEPATH=" + spec["python"].command.path)
-
-        if "+legacy-unit" in spec:
-            args.append("-DNRN_DYNAMIC_UNITS_USE_LEGACY=ON")
-
-        if "+caliper" in spec:
-            args.append("-DNRN_ENABLE_PROFILING=ON")
-            args.append("-DNRN_PROFILER=caliper")
-            args.append("-DCORENRN_ENABLE_CALIPER_PROFILING=ON")
-
-        # cmake options for coreneuron
-        if spec.satisfies("+coreneuron"):
-            options = [
-                "-DCORENRN_ENABLE_SPLAYTREE_QUEUING=ON",
-                "-DCORENRN_ENABLE_TIMEOUT=OFF",
-                "-DCORENRN_ENABLE_OPENMP=%s" % ("ON" if "+openmp" in spec else "OFF"),
-                "-DCORENRN_ENABLE_LEGACY_UNITS=%s" % ("ON" if "+legacy-unit" in spec else "OFF"),
-                "-DCORENRN_ENABLE_UNIT_TESTS=%s" % ("ON" if "+tests" in spec else "OFF"),
-            ]
+    # used for debugging purposes
+    variant(
+        "prcellstate",
+        default=False,
+        description="Enable tracking of voltage and conductivity with prcellstate on CoreNEURON",
+    )
 
-            nmodl_options = "codegen --force"
-            options.append("-DCORENRN_NMODL_FLAGS=%s" % nmodl_options)
+    # used furing development by core team
+    variant(
+        "sanitizers",
+        default="None",
+        description="Enable runtime sanitizers",
+        multi=True,
+        values=("None", "address", "leak", "undefined"),
+    )
 
-            if spec.satisfies("+gpu"):
-                nvcc = which("nvcc")
-                options.append(self.define("CMAKE_CUDA_COMPILER", nvcc))
-                options.append(self.define("CORENRN_ENABLE_GPU", True))
+    # standard deployment uses submodule to avoid compatibility issues
+    depends_on("nmodl", when="+coreneuron")
+    depends_on("libsonata-report", when="@9:+report+coreneuron")
 
-            args.extend(options)
 
-        # cache performance for build neuronsimulator/nrn/pull/1574
-        if spec.satisfies("@8.2:"):
-            args.append("-DNRN_AVOID_ABSOLUTE_PATHS=ON")
+    def cmake_args(self):
+        args = super(Neuron, self).cmake_args()
+        spec = self.spec
 
-        # enable tests that are selected via variant
+        # extra optimisation specific option added
+        nmodl_options = "codegen --force"
+        if spec.satisfies("+sympy"):
+            nmodl_options += " sympy --analytic"
+        if spec.satisfies("+sympyopt"):
+            nmodl_options += " --conductance --pade --cse"
+        args.append("-DCORENRN_NMODL_FLAGS=%s" % nmodl_options)
```

Comment:
`self.define`...

Reply:
@matz-e : could you check if I have updated all `-D..` correctly with `self.define` or equivalent?